nautobot/extras/secrets/__init__.py

```diff
@@ -31,6 +33,7 @@ def ParametersForm(self):
 
     @classmethod
     @abstractmethod
+    @unsafe
     def get_value_for_secret(cls, secret, obj=None, **kwargs):
```

Comment:
Does this also need `do_not_call_in_templates` or does that only apply to instance methods?

Reply:
This one (and the two below) I mostly added out of paranoia / belt-and-suspenders thinking. Unlike model instances, these classes *shouldn't* generally be exposed/accessible in Jinja templates... that said, there's probably a good argument for being consistent. I'll add them (and make the corresponding change in the 2.4.10 release PR). Thanks.